### ISSUE TITLE
how can i do smtp-serve ehlo command

### ISSUE THREAD

**Issue by linejine** [andris9/smtp-server#46]
i made a test program with smtp-server, smtp-connection
i ran the smtp-server and run the smtp-connection.
i encountered error on the smtp-connection.
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: Invalid greeting from server:
421 user-PC You talk too soon: 421 user-PC You talk too soon
    at SMTPConnection._actionGreeting (C:\Users\kjin\AppData\Roaming\npm\node_modules\smtp-connection\lib\smtp-connection.js:718:23)
    at SMTPConnection._processResponse (C:\Users\kjin\AppData\Roaming\npm\node_modules\smtp-connection\lib\smtp-connection.js:604:16)
    at SMTPConnection._onData (C:\Users\kjin\AppData\Roaming\npm\node_modules\smtp-connection\lib\smtp-connection.js:431:10)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)

Process finished with exit code 1

so, i think that it appears because of smtp-server ehlo command.
so how can i do ehlo command in smtp-server?

**Comment by andris9** [comment]
See the other response https://github.com/nodemailer/smtp-connection/issues/40

**Comment by ParadoXxGER** [comment]
@andris9 The resource you linked isn't available anymore, could you please let us know whats the solution on this? Thank you!

(I'm getting the same "421 You talk too soon" error)

**Comment by phil-warner** [comment]
@nhh did you figure this one out?

**Comment by phil-warner** [comment]
For others who stumble on this page, I got this error because I called the login beneath the connection.connect() function, rather than in it's callback.

**Comment by vikky1992** [comment]
> @andris9 The resource you linked isn't available anymore, could you please let us know whats the solution on this? Thank you!
> 
> (I'm getting the same "421 You talk too soon" error)

I am getting the same error. did u figure out a way?

**Comment by vikky1992** [comment]
I have cofigured the server as below.

```
SMTPServerOptions = { 
  secure: false,
  hideSTARTTLS:true,
  authOptional: true,
  debug: true,
  logger: true,
  onAuth,
  onData
}

if(conf.tls) {
  SMTPServerOptions.ca = fs.readFileSync('./certificates/chain.pem', 'ascii')
  SMTPServerOptions.key = fs.readFileSync('./certificates/privkey.pem','ascii')
  SMTPServerOptions.cert = fs.readFileSync('./certificates/cert.pem','ascii')
}
//creating new SMTP object
const server = new SMTPServer(SMTPServerOptions);

server.on('error', err => {
  error(err)
  throw err 
});

server.listen(conf.server_port);
```

I get the following error when the mail submission agent is used with STARTTLS option.
```
5|producer  | [2020-10-09 07:28:52] DEBUG [#ff7cqlwi7rat6z2k] C: EHLO qa.mydomain.com     

5|producer  | [2020-10-09 07:28:52] DEBUG [#ff7cqlwi7rat6z2k] S: 421 mydomain.com You talk too soon      

5|producer  | [2020-10-09 07:28:52] INFO  [#ff7cqlwi7rat6z2k] Connection closed to 91.198.201.301    
```
